src/Forms/GoToLine.cs

```diff
@@ -85,10 +85,7 @@ private void TextBox1KeyDown(object sender, KeyEventArgs e)
                     e.KeyCode == Keys.End ||
                     (e.KeyValue >= 96 && e.KeyValue <= 105))
                 {
-                }
-                else if (e.KeyData == (Keys.Shift | Keys.Home) || e.KeyData == (Keys.Shift | Keys.End) ||

```

Comment:
https://github.com/ivandrofly/subtitleedit/blob/patch-3.0/src/Forms/GoToLine.cs#L70
Will cover the second condition which would capture the Shift+Home...